byway/src/main/webapp/track-interests.js

```diff
@@ -12,13 +12,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let interestsChosen = new Set();
+let tripQuery;
+
 if(document.readyState === 'loading') {
-  document.addEventListener('DOMContentLoaded', loadButtons);
+  document.addEventListener('DOMContentLoaded', loadContent);
 } else {
+  loadContent();
+}
+
+/**
+ * Loads the content of the page by loading buttons
+ * and the form tracking buttons selected.
+ */
+function loadContent() {
+  loadId();
   loadButtons();
+  loadForm();
 }
 
-let interestsChosen = new Set();
+/* Parse url to retrieve the trip id. */
+function loadId() {
+  let queryString = location.search.substring(1);
+  let separatedData = queryString.split("&");
+  tripQuery = separatedData[0];
+}
+
+/**
+ * Sets properties to the interest form on the html
+ * page. Handles submit event to keep user on the
+ * same page and send information.
+ */
+function loadForm() {
+  let interestsForm = document.getElementById('interests-form');
+  interestsForm.addEventListener('submit', (event) => {
+    event.preventDefault();
+    let interestsAsJSONString = JSON.stringify(Array.from(interestsChosen));
+    let formData = new FormData();
+    formData.append("interests", interestsAsJSONString);
+    fetch('/api/places?' + tripQuery,
+      {method: 'POST', body: formData});
```

Comment:
Rather than writing out the logic for these kinds of fetches all over the place, consider making a common function that handles fetch with the correct path and query parameters. This will be especially useful if you start using a tripId for `GET` requests too.

For example,

```js
/**
 * Gets or updates the places on the server for the specified tripId. If no newInterests are specified, 
 * makes a get request to fetch the current places for the specified trip. Otherwise makes a post request
 * to update the places for the trip to the value specified. newInterests may be any type which can be
 * converted to an array by Array.from. It will be sent to the server as json in the post body. 
 */
function fetchPlaces(tripId, /* optional */ newInterests) {
  const url = '/api/places?' + new URLSearchParams({tripId}).toString();
  const fetchArgs = {method: 'GET'};
  if (newInterests !== undefined) {
    fetchArgs.method = 'POST';
    fetchArgs.body = JSON.stringify(Array.from(newInterests));
  }
  return fetch(url, fetchArgs).then(response => response.json());
}
```

Reply:
Updated